### ISSUE TITLE
panorama warping

### ISSUE THREAD

**Issue by yaodao-x** [fnzhan/Illumination-Estimation#6]
Hi, fnzhan:

          It is mentioned in the article, when extracting crops from panorama，you apply the same image warping as in [1] to each image.
         There is no open source code of [1], do you implement this method by yourself or ask the author for source code. Or you  don't apply this method while training the network.

Best wishes.

[1]Learning to predict indoor illumination from a single image. In: SIGGRAPH Asia (2017)

**Comment by fnzhan** [comment]
The warping operation is quite trivial, we try to implement it but give up soon. Later, we directly obtained the warped image from author of [1].

**Comment by yaodao-x** [comment]
OMG，you reply very quickly.  Thanks a lot.

**Comment by cyjouc** [comment]
Hi, fnzhan:
     would like share the method how to choose the best regression training model?the train.py script have not the model.val?
could you like guide the process？
best wishes!

**Comment by fnzhan** [comment]
You can divide a validation set, and compute the L1 distance between the gaussian maps reconstructed by predicted parameters and gt parameters as the validation accuracy. Then the best regression model can be selected.

**Comment by cyjouc** [comment]
TKS very much for your helpNow,I have some problems?Please guide me!
1、train.py script produce the summary.jpg，the left is original image,Is  the right predicted map? the right image is generated by the code(env_gt_pred = np.vstack((env_gt, env_pred)) ), can I understand the env_gt is gt parameters and env_pred is the predicted parameters?
However,from the predicted output image results, it feels that the predicted features are not quite right?
2、Do the angular error added the traing processing for the quantitative evaluation? would you like share the part of the code？

**Comment by fnzhan** [comment]
> TKS very much for your helpNow,I have some problems?Please guide me!
> 1、train.py script produce the summary.jpg，the left is original image,Is the right predicted map? the right image is generated by the code(env_gt_pred = np.vstack((env_gt, env_pred)) ), can I understand the env_gt is gt parameters and env_pred is the predicted parameters?
> However,from the predicted output image results, it feels that the predicted features are not quite right?
> 2、Do the angular error added the traing processing for the quantitative evaluation? would you like share the part of the code？

1. The training process may not be successful,  you can try to over-fit the model over a small set at first.
2.  The angular error is not added to the training process. angular error can be found in this paper 'DeepLight: Learning Illumination for Unconstrained Mobile Mixed Reality'.

**Comment by cyjouc** [comment]
Hi,github issus can not input the jpg image,help me quantitate the result?

![1](https://user-images.githubusercontent.com/65534092/124240601-795c7800-db4d-11eb-9a97-811cbb1a1929.jpg)

**Comment by cyjouc** [comment]
help me quantitate the result?
![1](https://user-images.githubusercontent.com/65534092/124240849-bfb1d700-db4d-11eb-979e-7f055def1f64.jpg)

**Comment by cyjouc** [comment]
Hi, fnzhan:
     training datas of the geprojector also divide 9,556 training pairs and  200 images as the val?

**Comment by cyjouc** [comment]
Hi, fnzhan:
the trained datas  of the geprojector also divided as the same of regression process?

**Comment by fnzhan** [comment]
> help me quantitate the result? ![1](https://user-images.githubusercontent.com/65534092/124240849-bfb1d700-db4d-11eb-979e-7f055def1f64.jpg)

Here are some expected training results:
![93_0](https://user-images.githubusercontent.com/21986728/144560306-c87a25bc-a0c4-449a-b872-9b7c4df075bf.jpg)
![93_1500](https://user-images.githubusercontent.com/21986728/144560325-92a51600-4ff5-49d1-ac46-a9670ed91745.jpg)

![93_2100](https://user-images.githubusercontent.com/21986728/144560344-b4599fc4-0307-40bf-806c-11f650164cc5.jpg)

**Comment by fnzhan** [comment]
> Hi, fnzhan: the trained datas of the geprojector also divided as the same of regression process?

Sorry for the late reply, there is too many things and deadlines. Definitely, the same dataset division for regression and generation.
